### PR TITLE
allow to build one snapshot twice

### PR DESCRIPTION
**Pull request by hicqu**
### What problem does this PR solve?

There is a bug about deleting a snapshot: the associated item in `DataKeyManager` is not cleaned. So that if the one snapshot is built twice, it will fail at this line:
```rust
rename_tmp_cf_file_for_send(...) {
    mgr.link(tmp_file_path, cf_file_path);
}
```
because `cf_file_path` is already in the key manager.

### What is changed and how it works?

Clean the item in `DataKeyManager` when deleting a cf file.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

```release-note
Avoid panic when building a snapshot twice if encryption enabled
```

### CODE REVIEW

**Review by hunterlxt — COMMENTED** [review]
What is the reason for directly operating the key manager? Why not delete the previous file normally

**Review by hunterlxt — APPROVED** [review]

File: components/raftstore/src/store/snap.rs [hunk 1]
```diff
@@ -1463,7 +1465,13 @@ impl SnapManagerCore {
...
             // It's ok that the cf file is moved but machine fails before `mgr.rename_file`
             // because without metadata file, saved cf files are nothing.
-            mgr.link_file(src, dst)?;
+            while let Err(e) = mgr.link_file(src, dst) {
+                if e.kind() == ErrorKind::AlreadyExists {
+                    mgr.delete_file(dst)?;
+                    continue;
+                }
+                return Err(e.into());
+            }
```
Comment (gengliqi) [inline components/raftstore/src/store/snap.rs:1474]: It seems not necessary. These snapshot files can be deleted in the `drop` function.

**Review by gengliqi — COMMENTED** [review]
Rest LGTM

**Review by gengliqi — APPROVED** [review]

**Pull request merged**